lib/utils.ts

```diff
@@ -199,155 +200,157 @@ export const Utils = new class {
-		}
-		return clone;
+		if (!skip) delete require.cache[path];
 	}
+}
+
+export function deepClone(obj: any): any {
+	if (obj === null || typeof obj !== 'object') return obj;
+	if (Array.isArray(obj)) return obj.map(prop => deepClone(prop));
+	const clone = Object.create(Object.getPrototypeOf(obj));
+	for (const key of Object.keys(obj)) {
+		clone[key] = deepClone(obj[key]);
+	}
+	return clone;
+}
 
-	levenshtein(s: string, t: string, l: number): number {
-		// Original levenshtein distance function by James Westgate, turned out to be the fastest
-		const d: number[][] = [];
+export function levenshtein(s: string, t: string, l: number): number {
+	// Original levenshtein distance function by James Westgate, turned out to be the fastest
+	const d: number[][] = [];
 
-		// Step 1
-		const n = s.length;
-		const m = t.length;
+	// Step 1
+	const n = s.length;
+	const m = t.length;
 
-		if (n === 0) return m;
-		if (m === 0) return n;
-		if (l && Math.abs(m - n) > l) return Math.abs(m - n);
+	if (n === 0) return m;
+	if (m === 0) return n;
+	if (l && Math.abs(m - n) > l) return Math.abs(m - n);
 
-		// Create an array of arrays in javascript (a descending loop is quicker)
-		for (let i = n; i >= 0; i--) d[i] = [];
+	// Create an array of arrays in javascript (a descending loop is quicker)
+	for (let i = n; i >= 0; i--) d[i] = [];
 
-		// Step 2
-		for (let i = n; i >= 0; i--) d[i][0] = i;
-		for (let j = m; j >= 0; j--) d[0][j] = j;
+	// Step 2
+	for (let i = n; i >= 0; i--) d[i][0] = i;
+	for (let j = m; j >= 0; j--) d[0][j] = j;
 
-		// Step 3
-		for (let i = 1; i <= n; i++) {
-			const si = s.charAt(i - 1);
+	// Step 3
+	for (let i = 1; i <= n; i++) {
+		const si = s.charAt(i - 1);
 
-			// Step 4
+		// Step 4
 			for (let j = 1; j <= m; j++) {
-				// Check the jagged ld total so far
-				if (i === j && d[i][j] > 4) return n;
+			// Check the jagged ld total so far
+			if (i === j && d[i][j] > 4) return n;
 
-				const tj = t.charAt(j - 1);
-				const cost = (si === tj) ? 0 : 1; // Step 5
+			const tj = t.charAt(j - 1);
+			const cost = (si === tj) ? 0 : 1; // Step 5
 
-				// Calculate the minimum
-				let mi = d[i - 1][j] + 1;
-				const b = d[i][j - 1] + 1;
-				const c = d[i - 1][j - 1] + cost;
+			// Calculate the minimum
+			let mi = d[i - 1][j] + 1;
+			const b = d[i][j - 1] + 1;
+			const c = d[i - 1][j - 1] + cost;
 
-				if (b < mi) mi = b;
-				if (c < mi) mi = c;
+			if (b < mi) mi = b;
+			if (c < mi) mi = c;
 
-				d[i][j] = mi; // Step 6
-			}
+			d[i][j] = mi; // Step 6
 		}
-
-		// Step 7
-		return d[n][m];
 	}
 
-	waitUntil(time: number): Promise<void> {
-		return new Promise(resolve => {
-			setTimeout(() => resolve(), time - Date.now());
-		});
-	}
-};
+	// Step 7
+	return d[n][m];
+}
+
+export function waitUntil(time: number): Promise<void> {
+	return new Promise(resolve => {
+		setTimeout(() => resolve(), time - Date.now());
+	});
+}
+
+// backwards compatibility
+export const Utils = module.exports;
```

Comment:
I'd prefer not to mix CommonJS and ES modules. It's probably better to just name every function again here.

Reply:
Alright.